priam/src/main/java/com/netflix/priam/restore/PostRestoreHook.java

```diff
@@ -0,0 +1,158 @@
+package com.netflix.priam.restore;
+
+import com.netflix.priam.IConfiguration;
+import com.netflix.priam.scheduler.NamedThreadPoolExecutor;
+import com.netflix.priam.utils.RetryableCallable;
+import com.netflix.priam.utils.Sleeper;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An implementation of IPostRestoreHook. Kicks off a child process for post restore hook using ProcessBuilder; uses heart beat monitor to monitor progress of the sub process
+ * and uses a file lock to pass the active state to the sub process
+ */
+public class PostRestoreHook implements IPostRestoreHook {
+    private static final Logger logger = LoggerFactory.getLogger(PostRestoreHook.class);
+    private final IConfiguration config;
+    private final Sleeper sleeper;
+    private final int ThreadWaitTimeInMs = 60000;
+    private final int HeartBeatTimeOutInMs = 60000;
+    private final int SubProcessTimeOutInDays = 2;
```

Comment:
Does it make sense to make this configurable?

Reply:
Also, once done, can you update the wiki to reflect the changes.

Reply:
I like the idea of making the subprocesstimeout configurable - comes in handy when restore data size varies